android/automation.md

```diff
@@ -0,0 +1,175 @@
+* Execute codecov for monitoring our code coverage
+
+To see a travis configuration for your project, see `<project-root>/.travis.yml` (here's an
+[example in focus-android][travis yml]).
+
+You can also find build logs on the travis website, e.g. for focus-android: https://travis-ci.org/mozilla-mobile/focus-android
+
+# Code quality tools
+
+For Java/Kotlin projects, we like to run the following tools:
+* findbugs
+* PMD
+* Android lint
+* checkstyle
+* [detekt](https://github.com/arturbosch/detekt)
+* [ktlint](https://github.com/shyiko/ktlint)
+
+To see what tools are currently run in automation for your project, see
+the project's `<project-root>/.taskcluster.yml`
+(here's an [example in focus-android][tc yml tools]).
+
+# Third-party services
+
+## Sputnik
+
+Sputnik (riggered from travis) runs findbugs, PMD and checkstyle on code attached to pull requests and posts in-line comments whenever violations are found.
+
+* https://sputnik.ci
+
+## Codecov
+
+Codecov evaluates the code coverage reports generated by our build and tracks the coverage over time. It warns when merging pull requests would lead to dropping the coverage below a specified threshold.
+
+* https://codecov.io/gh
+* [focus-android code coverage over time](https://codecov.io/gh/mozilla-mobile/focus-android/branch/master)
+
+# Automation tasks
+
+## Pull request pipeline
+
+![](https://i.imgur.com/SJ9zSs8.png)
+
+## Release pipeline
+
+Eventually we want to automate the whole release process pipeline in order to have "one click" releases without much workload for the engineering team.
+
+### ✅ Active: Release builds
+
+Whenever a release is created on GitHub we build releases versions of Focus and Klar on taskcluster.
+
+* [Configuration](https://github.com/mozilla-mobile/focus-android/blob/master/.taskcluster.yml#L91)
+
+### 📓 Planned: Signing
+
+Signing requires access to Mozilla's signing certificates. In addition to that a signing task needs to be able to validate the origin of the APK ("Can I trust the task that built the APK?") to avoid signing random tasks. To get this task up and running we need help from the release engineering team.
+
+Currently this is done manually by the release engineering team after we open a signing request bug on Bugzilla.
+
+* [Chain of Trust](http://scriptworker.readthedocs.io/en/latest/chain_of_trust.html)
+* [pushapkscript](https://github.com/mozilla-releng/pushapkscript)
+
+### 📓 Planned: Upload to archive.mozilla.org
+
+We want to publish every release on archive.mozilla.org. Currently we upload those builds manually. In automation this will be done by  the beetmover_scriptworker.
+
+Before we can automate this task we need to automate signing builds and have a chain of trust.
+
+* [beetmover script](https://github.com/mozilla-releng/beetmoverscript/)
+* [beetmover configuration for Fennec](https://github.com/mozilla-releng/beetmoverscript/blob/master/beetmoverscript/templates/fennec_nightly.yml)
+
+### 📓 Planned: Release signoff
```

Comment:
We decided to not do that. Instead of sign-offs via taskcluster we just upload to the Alpha channel and the "sign-off" is promoting the build on Google Play.

Reply:
Removed this section.